agent/testflinger_agent/agent.py

```diff
@@ -17,13 +17,24 @@
 import os
 from pathlib import Path
 import shutil
-import tarfile
 import tempfile
 
 from testflinger_agent.job import TestflingerJob
 from testflinger_agent.errors import TFServerError
 from testflinger_agent.config import ATTACHMENTS_DIR
 
+try:
+    # attempt importing a tarfile filter, to check if filtering is supported
+    from tarfile import data_filter
+
+    del data_filter
+except ImportError:
+    # import a patched version of `tarfile` that supports filtering
+    from . import tarfile_patch as tarfile
```

Comment:
It might be useful to add to the comment here something like "This can be removed once the minimum version of Python is X.Y.Z" (3.9? 3.12? 3.8.17?)

Reply:
Done. I actually had one and then removed it :grin: